File: src/android/lukuid-sdk/src/main/java/com/lukuid/sdk/internal/DeviceInfoCache.kt
// SPDX-License-Identifier: Apache-2.0
package com.lukuid.sdk.internal

import com.lukuid.sdk.DeviceInfo
import com.lukuid.sdk.TransportType
import java.util.concurrent.ConcurrentHashMap

internal class DeviceInfoCache {
    private val cache = ConcurrentHashMap<String, DeviceInfo>()

    fun get(transport: TransportType, transportId: String): DeviceInfo? = cache[key(transport, transportId)]

    fun put(transport: TransportType, transportId: String, info: DeviceInfo) {
        cache[key(transport, transportId)] = info
    }

    private fun key(transport: TransportType, transportId: String) =
        "${transport.wireName}:$transportId"
}

File: src/ios/LukuIDSDK/Sources/LukuIDSDK/DeviceInfoCache.swift
// SPDX-License-Identifier: Apache-2.0
import Foundation

actor DeviceInfoCache {
    private var cache: [String: DeviceInfo] = [:]

    func get(transport: TransportType, transportId: String) -> DeviceInfo? {
        cache[key(for: transport, id: transportId)]
    }

    func set(_ info: DeviceInfo) {
        cache[key(for: info.transport, id: info.transportId)] = info
    }

    private func key(for transport: TransportType, id: String) -> String {
        "\(transport.rawValue):\(id)"
    }
}
